### DISCUSSION TITLE
Enhance DataTable.Scenario(title) with function

### DISCUSSION THREAD

**Discussion by eJayYoung** [codeceptjs/CodeceptJS#3102] [Ideas]
hi, I have a `Data(xxx).Scenario() ` usage , which I thougth the [replaceTitle](https://github.com/codeceptjs/CodeceptJS/blob/3.x/lib/data/context.js#L64) can be more enhanced for user to custom their Scenario title

from now the usage
```js
Data([{
  title: 'staff login',
  account: 'staff',
}, {
  title: 'manager login',
  account: 'manager'
}]).Scenario('login', () => {})

// console output
✔ login | {"title": "staff login", "account": "staff"}
✔ login | {"title": "manager login", "account": "manager"}
```

what's I expect usage
```js
Data([{
  title: 'staff login',
  account: 'staff',
}, {
  title: 'manager login',
  account: 'manager'
}]).Scenario(({ title }) => `${title} success`, () => {})

// console output
✔ staff login success
✔ manager login success
```

so, just support the first argument `title` as a function can be more scalable
I can give the PR in this weekend
look forward to your reply
Best regards!

**Comment by Yasuslik** [comment]
I have very big data in data provider, and i to need this...
do you create PR?

**Comment by kobenguyent** [comment]
done here https://github.com/codeceptjs/CodeceptJS/pull/4030
